x-pack/plugins/task_manager/server/MONITORING.md

```diff
@@ -0,0 +1,242 @@
+# Task Manager Monitoring
+
+Task Manager has an internal monitoring mechanism in which keeps track of a variety of metrics which are exposed via a `health` api endpoint and Kibana Server Log debug messaging.
+
+## Exposed Metrics
+There are three different sections to the stats returned by the `health` api.
+- `configuration`: Summarizes Task Manager's current configuration.
+- `workload`: Summarizes the workload in the current deployment.
+- `runtime`: Tracks Task Manager's performance.
+
+### Configuring the Stats
+There are three new configurations:
+
+- `xpack.task_manager.monitored_stats_required_freshness` - The _required freshness_ of critical "Hot" stats, which means that if key stats (last polling cycle time, for example) haven't been refreshed within the specified duration, the `_health` endpoint and service will report an `Error` status. By default this is inferred from the configured `poll_interval` and is set to `poll_interval` plus a `1s` buffer.
```

Comment:
A one second buffer seems too small to me - I'd be a little afraid of a Kibana under load still working "fine", but exceeding that, moving to an Error.  I wonder if we should have two values, perhaps computed from one config value - a warning and an error.  Where warning was using the config value, error used 2x the config value.  Presumably the value can't be less than `poll_interval`, perhaps should not be lower than whatever "default we use (`poll_interval` + `1s`).

Reply:
Yeah, that's fair - this is just an initial stab.
As mentioned on the issue itself - I'd like some input on where we draw the line between Error and Warning.

Reply:
Yeah it does get interesting. Seems it could follow what the query timeout would be. Some queries against Elasticsearch can take a while to run and should be perfectly normal (part of make it slow project).